ludwig/features/sequence_feature.py

```diff
@@ -229,385 +220,100 @@ def __init__(self, feature):
-
-            last_predictions = tf.gather_nd(
-                predictions_sequence,
-                tf.stack(
-                    [tf.range(tf.shape(predictions_sequence)[0]),
-                     tf.maximum(
-                         predictions_sequence_length - 1 - additional,
-                         0
-                     )],
-                    axis=1
-                )
-            )
 
-            last_targets = tf.gather_nd(
-                targets,
-                tf.stack(
-                    [tf.range(tf.shape(predictions_sequence)[0]),
-                     tf.maximum(targets_sequence_length - 1 - additional, 0)],
-                    axis=1
-                )
-            )
+        logits = inputs[LOGITS]
 
-        return (
-            predictions_sequence,
-            predictions_sequence_scores,
-            predictions_sequence_length,
-            last_predictions,
-            probabilities_sequence,
-            targets_sequence_length,
-            last_targets,
-            eval_logits,
-            train_logits,
-            class_weights,
-            class_biases
+        probabilities= tf.nn.softmax(
+            logits,
+            name='probabilities_{}'.format(self.name)
         )
-
-    def sequence_metrics(
-            self,
-            targets,
-            targets_sequence_length,
-            last_targets,
-            predictions_sequence,
-            predictions_sequence_length,
-            last_predictions,
-            eval_loss
-    ):
-        with tf.variable_scope('metrics_{}'.format(self.feature_name)):
-            (
-                token_accuracy_val,
-                overall_correct_predictions,
-                rowwise_accuracy_val,
-                rowwise_correct_predictions
-            ) = masked_accuracy(
-                targets,
-                predictions_sequence,
-                targets_sequence_length,
-                self.feature_name
-            )
-            last_accuracy_val, correct_last_predictions = accuracy(
-                last_targets,
-                last_predictions,
-                self.feature_name
-            )
-            edit_distance_val, mean_edit_distance = edit_distance(
-                targets,
-                targets_sequence_length,
-                predictions_sequence,
-                predictions_sequence_length,
-                self.feature_name
-            )
-            perplexity_val = perplexity(eval_loss)
-
-        return (
-            correct_last_predictions,
-            last_accuracy_val,
-            overall_correct_predictions,
-            token_accuracy_val,
-            rowwise_correct_predictions,
-            rowwise_accuracy_val,
-            edit_distance_val,
-            mean_edit_distance,
-            perplexity_val
+        predictions = tf.argmax(
+            logits,
+            -1,
+            name='predictions_{}'.format(self.name),
+            output_type=tf.int32
```

Comment:
this may be int64 in some cases (when the vocabulary is bigger than a certain amount. I'm not sure if TF is smart enough  to return a int64 if the number of dimensions in `c` in the tensor is higher than a certain amount. Also, when computing the loss and the metrics, I would make sure that both predictions and groundtruth are the same precision or that the loss and metrics functions work well with ints of different precision. This is because when there's a small vocabulary the ints may be int8 or int16, something I did to save both disk space and memory.

Reply:
I believe you're recommending that I add some tests to determine how large a encode value could be for an element in the sequence.  Based on this maximum value, I would use  tf.int8, tf.int16, tf.int32, tf.int64 as needed.  I addition, I should make sure the same integer type should be applied to both targets and predictions.  

Is this a correct understanding?